### PR TITLE
docs: callout the need to restart pods to pickup service accounts

### PR DESCRIPTION
**Pull request by dennispan**
Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

**Review by blakepettersson — APPROVED** [review]

File: docs/operator-manual/declarative-setup.md [hunk 1]
```diff
@@ -676,8 +676,10 @@ extended to allow assumption of multiple roles, either as an explicit array of r
   }
 ```
 
-Example service account configs for `argocd-application-controller` and `argocd-server`. Note that once the annotations
-have been set on the service accounts, both the application controller and server pods need to be restarted.
+Example service account configs for `argocd-application-controller` and `argocd-server`.
+
+!!! warning
+    Once the annotations have been set on the service accounts, both the application controller and server pods need to be restarted.
```
Comment (blakepettersson) [inline docs/operator-manual/declarative-setup.md:682]: I think for it to render as a warning box, it should rather be like this

```suggestion
> [!WARNING]
>  Once the annotations have been set on the service accounts, both the application controller and server pods need to be restarted.
```

Comment (dennispan) [reply docs/operator-manual/declarative-setup.md:682]: I used a few other warning messages in Argo CD doc as reference. Actually I cannot find `[!WARNING]` being used

image
image

Comment (blakepettersson) [reply docs/operator-manual/declarative-setup.md:682]: Yeah you're right, was looking at Github Markdown syntax rather than readthedocs.

**Comment by blakepettersson** [comment]
@dennispan you need to signoff your commit: https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal

**Comment by dennispan** [comment]
@blakepettersson done

**Comment by blakepettersson** [comment]
Awesome, thanks @dennispan!

**Pull request merged**